src/main/java/org/jenkinsci/plugins/parallel_test_executor/PreviousTestResultLookup.java

```diff
@@ -0,0 +1,61 @@
+package org.jenkinsci.plugins.parallel_test_executor;
+
+import com.google.common.base.Predicate;
+import hudson.ExtensionPoint;
+import hudson.model.Run;
+import hudson.tasks.test.TestResult;
+import org.jenkinsci.plugins.variant.OptionalExtension;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepStartNode;
+import org.jenkinsci.plugins.workflow.flow.FlowExecution;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.graphanalysis.DepthFirstScanner;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.support.steps.StageStep;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public abstract class PreviousTestResultLookup implements ExtensionPoint {
```

Comment:
A public API with no Javadoc?

For that matter, why is this an `ExtensionPoint` at all? You are never looking for its `ExtensionList`!

And since you are for now implementing it inside the plugin (ick), it does not really need to be `public` at all.

Reply:
Yeah, I got overeager, this isn’t documented yet and needs to be. It’s an `ExtensionPoint` basically so that I can make the implementation an optional extension. Alternative approaches welcome. =)